Algorithm/QCAlgorithm.Trading.cs

```diff
@@ -957,48 +957,40 @@ public decimal CalculateOrderQuantity(Symbol symbol, decimal target)
             decimal marginRequired;
             decimal orderValue;
             decimal orderFees;
-            var feeToPriceRatio = 0;
+            var feeToPriceRatio = 0m;
 
             // compute the initial order quantity
-            decimal orderQuantity = targetOrderValue / unitPrice;
+            var orderQuantity = targetOrderValue / unitPrice;
 
-            if (orderQuantity % security.SymbolProperties.LotSize != 0)
-            {
-                orderQuantity = orderQuantity - (orderQuantity % security.SymbolProperties.LotSize);
-            }
-
-            var iterations = 0;
+            // rounding off Order Quantity to the nearest multiple of Lot Size
+            orderQuantity -= orderQuantity % security.SymbolProperties.LotSize;
 
             do
             {
-                // decrease the order quantity
-                if (iterations > 0)
-                {
-                    // if fees are high relative to price, we reduce the order quantity faster
-                    if (feeToPriceRatio > 0)
-                        orderQuantity -= feeToPriceRatio;
-                    else
-                        orderQuantity--;
-                }
+                // reduce order quantity by feeToPriceRatio, since it is faster than by lot size
+                // if it becomes nonpositive, return zero
+                orderQuantity -= feeToPriceRatio;
+                if (orderQuantity <= 0) return 0;
 
                 // generate the order
                 var order = new MarketOrder(security.Symbol, orderQuantity, UtcTime);
                 orderValue = order.GetValue(security);
                 orderFees = security.FeeModel.GetOrderFee(security, order);
-                feeToPriceRatio = (int)(orderFees / unitPrice);
+
+                // calculate feeToPriceRatio, if lower than lot size, use lot size instead
```

Comment:
This comment describes exactly the next few lines of code, however it doesn't tell me *why* the next few lines of code are there. I think what this is really trying to do is find an incremental delta value for the next iteration step. This algorithm is basically a newton rhapson method on an undefined function (due to fee model).

Also, the removal of the `iterations` variable from this loop is somewhat concerning. I'm curious how many iterations the algorithm needs now. Previously I think most requests only required ~2 or ~3 iterations to complete (max of 2 for linear fee model with #shares).

Reply:
I removed the iterations, because it was not being really used.
You are totally right, I ran the tests and it never needed more than 2 iterations.